content/en/docs/Writing policies/external-data-sources.md

```diff
@@ -0,0 +1,440 @@
+
+* `/apis/{GROUP}/{VERSION}/{RESOURCETYPE}`: get a collection of resources
+* `/apis/{GROUP}/{VERSION}/{RESOURCETYPE}/{NAME}`: get a resource 
+
+For namespaced resources, to get a specific resource by name or to get all resources in a namespace, the namespace name must also be provided as follows: 
+
+* `/apis/{GROUP}/{VERSION}/namespaces/{NAMESPACE}/{RESOURCETYPE}`: get a collection of resources in the namespace
+* `/apis/{GROUP}/{VERSION}/namespaces/{NAMESPACE}/{RESOURCETYPE}/{NAME}`: get a resource in a namespace
+
+For historic resources, the Kubernetes Core API is available under `/api/v1`. For example, to query all namespaces the path `/api/v1/namespaces` is used.
+
+The Kubernetes API groups are defined in the [API reference documentation](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#-strong-api-groups-strong-) and can also be retrieved via the `kubectl api-resources` command shown below:
+
+
+```sh
+λ kubectl api-resources
+NAME                              SHORTNAMES   APIGROUP                       NAMESPACED   KIND
+bindings                                                                      true         Binding
+componentstatuses                 cs                                          false        ComponentStatus
+configmaps                        cm                                          true         ConfigMap
+endpoints                         ep                                          true         Endpoints
+events                            ev                                          true         Event
+limitranges                       limits                                      true         LimitRange
+namespaces                        ns                                          false        Namespace
+nodes                             no                                          false        Node
+persistentvolumeclaims            pvc                                         true         PersistentVolumeClaim
+
+...
+
+
+```
+
+The `kubectl api-versions` command prints out the available versions for each API group. Here is a sample:
+
+```sh
+$ kubectl api-versions
+admissionregistration.k8s.io/v1
+admissionregistration.k8s.io/v1beta1
+apiextensions.k8s.io/v1
+apiextensions.k8s.io/v1beta1
+apiregistration.k8s.io/v1
+apiregistration.k8s.io/v1beta1
+apps/v1
+authentication.k8s.io/v1
+authentication.k8s.io/v1beta1
+authorization.k8s.io/v1
+authorization.k8s.io/v1beta1
+autoscaling/v1
+autoscaling/v2beta1
+autoscaling/v2beta2
+batch/v1
+   
+...
+
+```
+
+You can use these command together to find the URL path for resources, as shown below:
+
+
+{{% alert title="Tip" color="info" %}}
+
+To find the API group and version for a resource use `kubectl api-resources` to find the group and then `kubectl api-versions` to find the available versions.
+
+This example find the group of `deployment` resources and then queries the version:
```

Comment:
"This example _finds_ the group..."

Reply:
"...group of _Deployment_ resources..."

Reply:
You could also use a one-liner like such that gets both parts done:

`kubectl api-versions | grep $(kubectl api-resources | grep deployment | tr -s ' ' | cut -d' ' -f3)`

Reply:
> 
> 
> You could also use a one-liner like such that gets both parts done:
> 
> `kubectl api-versions | grep $(kubectl api-resources | grep deployment | tr -s ' ' | cut -d' ' -f3)`

This doesn't work for resources with multiple groups and those with some columns omitted.

Reply:
yeah, I tried the one-liner as well and them ran into issues and abandoned it!